analyses/molecular-subtyping-LGAT/04-LGAT-compile-subtypes.Rmd

```diff
@@ -0,0 +1,290 @@
+                     guess_max = 10000) %>%
+  filter( broad_histology == "Low-grade astrocytic tumor") %>%
+  select(Kids_First_Biospecimen_ID,
+         Kids_First_Participant_ID,
+         sample_id,
+         experimental_strategy,
+         cancer_predispositions)
+
+lgat_wgs <- lgat_manifest %>%
+  filter(experimental_strategy != "RNA-Seq") %>%
+  select( -experimental_strategy, -cancer_predispositions)
+
+lgat_rna <- lgat_manifest %>%
+  filter(experimental_strategy == "RNA-Seq") %>%
+  select( -experimental_strategy, -cancer_predispositions)
+
+all_lgat_matched <- full_join(lgat_wgs, lgat_rna,
+                         by = c("Kids_First_Participant_ID",
+                                "sample_id"),
+                         suffix = c("_DNA", "_RNA")) %>%
+  select(starts_with("Kids_First"), sample_id)
+
+# get subset folder
+subset_dir <- file.path(
+  root_dir,
+  "analyses",
+  "molecular-subtyping-LGAT",
+  "lgat-subset"
+)
+
+# result Dir
+resultDir <- file.path(
+  root_dir,
+  "analyses",
+  "molecular-subtyping-LGAT",
+  "results"
+)
+
+# SNV based annotation
+lgat_snv <- read_tsv(file.path(subset_dir,"LGAT_snv_subset.tsv")) 
+
+# CNV based annotation
+lgat_cnv <- read_tsv(file.path(subset_dir,"LGAT_cnv_subset.tsv")) %>%
+  # To-do add to CNV subset script
+  rename(Kids_First_Biospecimen_ID=lgat_wgs_biospecimen_ids)
+
+# Fusion base annotation
+lgat_fus <- read_tsv(file.path(subset_dir,"LGAT_fusion_subset.tsv")) 
+
+```
+
+
+### LGG, NF1
+This subtype is characterized by germline NF1 variants. We do not have germline data within OpenPBTA, but we do have information about cancer predispositions. This group can be annotated if NF-1 is present within the cancer_predispositions field of the pbta-histologies.tsv file. We should also annotate somatic loss of NF1 via either missense, nonsense mutation or copy loss.
+```{r}
+nf1_snv <- lgat_snv %>%
+  filter(NF1_mut=="Yes") %>%
+  mutate(molecular_subtype = "LGG, NF1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+nf1_germline <- lgat_manifest %>% 
+  # to keep non-RNA files only since we will match the RNA 
+  # as Kids_First_Biospecimen_ID_RNA column at a later step
+  filter(experimental_strategy != "RNA-Seq") %>%
+  # keeping "NF1 and NF2" in cancer_predisposition
+  filter(grepl("NF", cancer_predispositions)) %>%
+  mutate(molecular_subtype = "LGG, NF1") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+nf1_biospecimen <- bind_rows(nf1_snv,
+                             nf1_germline) %>%
+  unique()
+  
+```
+
+
+
+
+
+### LGG, KIAA1549-BRAF
+
+This subtype contains KIAA1549-BRAF fusion
+```{r}
+kiaabraf_biospecimen <- lgat_fusion %>%
+  filter(KIAA_BRAF_fus=="Yes") %>%
+  mutate(molecular_subtype = "LGG, KIAA1549-BRAF") %>%
+  select("Kids_First_Biospecimen_ID","molecular_subtype") 
+
+```
+
+### LGG, BRAF V600E
+
+This subtype contains BRAF V600E or V599 SNV or non-canonical BRAF alterations such as p.V600ins or p.D594N
```

Comment:
Will domain experts know that "non-canonical BRAF alterations such as p.V600ins or p.D594N" could fall under the `LGG, BRAF V600E` umbrella?

Reply:
In this [LGG review](https://www.sciencedirect.com/science/article/abs/pii/S1535610820301513) they gathered the 2 sites in BRAF p.V600E subtype so we also added to the list for other BRAF SNVs should I link it in the notebook? 

Here's the text from the paper: 
>Together, KIAA1549-BRAF (n = 166), BRAF p.V600E (n = 79), and germline NF1 mutations (n = 79) accounted for 68% (n = 324) of tumors (Figures 2A–2C). Rare alterations accounted for an additional 17% of cases. These included non-canonical BRAF alterations, such as fusions partnered with genes others than KIAA1549 (n = 4), 2 insertion events at position 600 (p.V600ins) and 1 SNV at position 594 (p.D594N) (Figures 2A)
![image](https://user-images.githubusercontent.com/34580719/105747905-621abe00-5f0f-11eb-88b8-d5150eeb119b.png)

Reply:
I guess I'm wondering if something like "LGG, Other BRAF alteration" would be more appropriate in those cases, but that would probably imply fusions other than the canonical BRAF fusion which is captured by "other MAPK"

Reply:
ah right!  Yeah it would be clearer to add the other BRAF snvs to "other MAPK", checking the script for if we have these other snvs in BRAF pV600E subtype

Reply:
Seems like in the 01 script I only looked for 599, 600 or SNVs overlapping tyrosine kinase domain: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/f85831fe7e405b1c6b6c43274cabcc796b7bd3b1/analyses/molecular-subtyping-LGAT/01-subset-files-for-LGAT.Rmd#L146-L156

```r
consensusMutationSubset[grep("BRAF",consensusMutationSubset$Hugo_Symbol),"HGVSp_Short"] %>% unique()
# A tibble: 4 x 1
  HGVSp_Short    
  <chr>          
1 p.V600E        
2 p.T599dup      
3 p.A598_T599insV
4 p.V600= 
```

I'm updating the comment in the notebook for BRAF V600E subtype

Reply:
> I guess I'm wondering if something like "LGG, Other BRAF alteration" would be more appropriate in those cases, but that would probably imply fusions other than the canonical BRAF fusion which is captured by "other MAPK"

I had put them in the v600e group per @adamcresnick's request.

Reply:
Okay, I think the documentation needs to be very clear about the criteria.

Reply:
Agree!